src/metadata/evaluator/filter/Filters.js

```diff
@@ -77,9 +78,10 @@ export default class Filters {
      * @param onChange
      * @returns {*}
      */
-    static entityFilter(propertyMetadata, model, keyPrefix, metadataEvaluator, reduxProps, onChange) {
+    static entityFilter(propertyMetadata, model, keyPrefix, metadataEvaluator, reduxProps, onChange, globalScope) {
         if (!propertyMetadata) throw new Error('metadata is required');
         if (!model) throw new Error('model is required');
+        if( !globalScope ) globalScope = model
```

Comment:
@nikhilarunw ohh! I understand, great job! I will clone your branch to test that the proposed feature works well. It's sure that I will merge it by the afternoon! 

Thanks again for your contribution! 😄